crates/interpreter/src/gas/calc.rs

```diff
@@ -161,214 +45,13 @@ pub const fn initcode_cost(len: usize) -> u64 {
-        ISTANBUL_SLOAD_GAS
-    } else {
-        SSTORE_RESET
-    }
-}
-
-/// Dynamic gas cost for sstore.
-#[inline]
-pub const fn sstore_cost_dynamic(spec_id: SpecId, vals: &SStoreResult, is_cold: bool) -> u64 {
-    sstore_cost(spec_id, vals, is_cold) - sstore_cost_static(spec_id)
-}
-
-/// Static gas cost for sstore.
-#[inline]
-pub const fn static_sstore_cost(spec_id: SpecId) -> u64 {
-    if spec_id.is_enabled_in(SpecId::BERLIN) {
-        WARM_STORAGE_READ_COST
-    } else if spec_id.is_enabled_in(SpecId::ISTANBUL) {
-        ISTANBUL_SLOAD_GAS
-    } else {
-        SSTORE_RESET
-    }
-}
-
-/// Dynamic gas cost for sstore.
-#[inline]
-pub const fn dyn_sstore_cost(spec_id: SpecId, vals: &SStoreResult, is_cold: bool) -> u64 {
-    sstore_cost(spec_id, vals, is_cold) - static_sstore_cost(spec_id)
-}
-
-/// `SSTORE` opcode cost calculation.
-#[inline]
-pub const fn sstore_cost(spec_id: SpecId, vals: &SStoreResult, is_cold: bool) -> u64 {
-    if spec_id.is_enabled_in(SpecId::BERLIN) {
-        // Berlin specification logic
-        let mut gas_cost = istanbul_sstore_cost::<WARM_STORAGE_READ_COST, WARM_SSTORE_RESET>(vals);
-
-        if is_cold {
-            gas_cost += COLD_SLOAD_COST;
-        }
-        gas_cost
-    } else if spec_id.is_enabled_in(SpecId::ISTANBUL) {
-        // Istanbul logic
-        istanbul_sstore_cost::<ISTANBUL_SLOAD_GAS, SSTORE_RESET>(vals)
-    } else {
-        // Frontier logic
-        frontier_sstore_cost(vals)
-    }
-}
-
-/// EIP-2200: Structured Definitions for Net Gas Metering
-#[inline]
-const fn istanbul_sstore_cost<const SLOAD_GAS: u64, const SSTORE_RESET_GAS: u64>(
-    vals: &SStoreResult,
-) -> u64 {
-    if vals.is_new_eq_present() {
-        SLOAD_GAS
-    } else if vals.is_original_eq_present() && vals.is_original_zero() {
-        SSTORE_SET
-    } else if vals.is_original_eq_present() {
-        SSTORE_RESET_GAS
-    } else {
-        SLOAD_GAS
-    }
-}
-
-/// Frontier sstore cost just had two cases set and reset values.
-#[inline]
-const fn frontier_sstore_cost(vals: &SStoreResult) -> u64 {
-    if vals.is_present_zero() && !vals.is_new_zero() {
-        SSTORE_SET
-    } else {
-        SSTORE_RESET
-    }
-}
-
-/// Static gas cost for selfdestruct.
-#[inline]
-pub const fn static_selfdestruct_cost(spec_id: SpecId) -> u64 {
-    // EIP-150: Gas cost changes for IO-heavy operations
-    if spec_id.is_enabled_in(SpecId::TANGERINE) {
-        5000
-    } else {
-        0
-    }
-}
-
-/// `SELFDESTRUCT` opcode cost calculation.
-#[inline]
-pub const fn dyn_selfdestruct_cost(spec_id: SpecId, res: &StateLoad<SelfDestructResult>) -> u64 {
-    let is_tangerine = spec_id.is_enabled_in(SpecId::TANGERINE);
-    let mut gas = 0;
-
-    // EIP-161: State trie clearing (invariant-preserving alternative)
-    let should_charge_topup = if spec_id.is_enabled_in(SpecId::SPURIOUS_DRAGON) {
-        res.data.had_value && !res.data.target_exists
-    } else {
-        !res.data.target_exists
-    };
```

Comment:
This logic is moved to selfdestruct instruction:

https://github.com/bluealloy/revm/blob/403f28f96b569d5d25966d43e9b9d1115965a084/crates/interpreter/src/instructions/host.rs#L379-L385